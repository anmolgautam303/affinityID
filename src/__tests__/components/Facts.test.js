import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import Facts from '../../components/facts';

configure({ adapter: new Adapter() })

describe('Facts', () => {
    const getRandomfact = jest.fn();
    let props = { getRandomfact: getRandomfact };
    let facts = shallow(<Facts {...props}/>);

    it('renders properly', () => {
        expect(Facts).toMatchSnapshot();
    });

    describe('user wants to get a random fact', () => {

        it('dispatches `getRandomfact()``', () => {
            facts.find('.btn').simulate('click')
            expect(props.getRandomfact).toHaveBeenCalledWith();
        });
    });
});