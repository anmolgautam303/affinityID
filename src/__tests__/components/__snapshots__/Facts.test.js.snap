// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facts renders properly 1`] = `[Function]`;
